src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -20,26 +19,33 @@ public class JsonDataSerializer : IDataSerializer
     {
         private static JsonDataSerializer instance;
 
-        private static JsonSerializer serializer;
+        private static JsonSerializer payloadSerializer;
+        private static JsonSerializer payloadSerializer2;
 
         /// <summary>
         /// Prevents a default instance of the <see cref="JsonDataSerializer"/> class from being created.
         /// </summary>
         private JsonDataSerializer()
         {
-            serializer = JsonSerializer.Create(
-                            new JsonSerializerSettings
-                                {
-                                    ContractResolver = new TestPlatformContractResolver(),
-                                    DateFormatHandling = DateFormatHandling.IsoDateFormat,
-                                    DateParseHandling = DateParseHandling.DateTimeOffset,
-                                    DateTimeZoneHandling = DateTimeZoneHandling.Utc,
-                                    TypeNameHandling = TypeNameHandling.None
-                                });
+            var jsonSettings = new JsonSerializerSettings
+            {
+                DateFormatHandling = DateFormatHandling.IsoDateFormat,
+                DateParseHandling = DateParseHandling.DateTimeOffset,
+                DateTimeZoneHandling = DateTimeZoneHandling.Utc,
+                TypeNameHandling = TypeNameHandling.None
+            };
+
+            payloadSerializer = JsonSerializer.Create(jsonSettings);
+            payloadSerializer2 = JsonSerializer.Create(jsonSettings);
+
+            payloadSerializer.ContractResolver = new TestPlatformContractResolver1();
+            payloadSerializer2.ContractResolver = new DefaultTestPlatformContractResolver();
```

Comment:
When v3 comes in, we need to change this code? Why not consider versioning the ContractResolvers suffixed with version number. The class name need not say what's default, let it be determined only in this file.